### PR TITLE
feat: Automatically grant user default roles

### PR DESCRIPTION
**Pull request by dgniewek**
Add option to automatically `GRANT` users `default_role` and `default_secondary_roles` .

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -102,3 +102,9 @@ variable "must_change_password" {
   type        = bool
   default     = true
 }
+
+variable "grant_default_roles" {
+  description = "Whether to grant defalut_role and default_secondary_roles to Snowflake User"
```
Comment (mariusz89016) [inline variables.tf:107]:
```suggestion
  description = "Whether to grant default_role and default_secondary_roles to Snowflake User"
```

File: main.tf [hunk 2]
```diff
@@ -52,3 +45,17 @@ resource "snowflake_user" "this" {
   rsa_public_key   = local.rsa_public_key
   rsa_public_key_2 = var.rsa_public_key_2
 }
+
+resource "snowflake_role_grants" "default_role" {
+  count = var.grant_default_roles && var.default_role != null ? 1 : 0
```
Comment (mariusz89016) [inline main.tf:50]: shouldn't we check also `module.this.enabled`?

File: main.tf [hunk 3]
```diff
@@ -52,3 +45,17 @@ resource "snowflake_user" "this" {
...
+
+  role_name = var.default_role
+  users     = [one(resource.snowflake_user.this[*].name)]
+}
+
+resource "snowflake_role_grants" "default_secondary_roles" {
+  for_each = var.grant_default_roles ? toset(var.default_secondary_roles) : []
```
Comment (mariusz89016) [inline main.tf:57]: same here: check for `module.this.enabled`? (IIUC that resource will be created even though an entire module is disabled, right?)

File: variables.tf (hunk 1 above)
Comment (dgniewek) [reply variables.tf]: Good catch - added

File: main.tf (hunk 2 above)
Comment (dgniewek) [reply main.tf]: Good catch - added

File: main.tf (hunk 3 above)
Comment (dgniewek) [reply main.tf]: Good catch - added

**Review by moss2k13 — APPROVED** [review]

**Pull request merged**